core/typedef.go

```diff
@@ -369,6 +375,8 @@ func (typeDef *TypeDef) ToInput() dagql.Input {
 		typed = DynamicID{typeName: typeDef.AsInterface.Value.Name}
 	case TypeDefKindVoid:
 		typed = Void{}
+	case TypeDefKindEnum:
+		typed = dagql.NewScalar[dagql.String](typeDef.AsEnum.Value.Name, dagql.String(""))
```

Comment:
Hm, why isn't this an `EnumObject` or similar? I think this might accidentally mean that we lose input validation (but I may be wrong).

When you write the tests, make sure to include tests with invalid enum values (see how we do that for scalars already). If they fail, it's likely this line at fault.

Reply:
I still got the input validation but it's done by the `ConvertToSDKInput` & `ConvertFromSDKResult` in `enum`

To use `EnumObject`, I must implement the `Input` interface but I already did it for `DynamicEnumValue` so I don't know.
I've been quite confused on this part honestly, about the  `Input` interface..

Reply:
Okay, so here's what I would try (may not actually be like this, but this would hugely simplify the structure).
- Remove the `DynamicEnumValue` interface - we should have a counterpart for the `EnumValues` struct called `DynamicEnumValues`. Both of these should implement `EnumValue` (the interface).
- Collapse all the functionality of `EnumObject` into `DynamicEnumValues` - I don't think we need both structs.

But I think this line is probably correct as-is - I think this is fine, but I still think we should try the above. I think that's what's causing a lot of the confusion.

Reply:
Okay I'm going to try that! Thanks for the hint

Reply:
Done! Thanks a lot